CMakeLists.txt

```diff
@@ -24,8 +25,37 @@ endif()
 
 string(REPLACE "-" "." gitver "${gitver}")
 
+option(BINARY_32BIT "compile 32bit version" OFF)
+option(DEBUG_OUTPUT "show extra debug output" OFF)
+option(DEBUG_SYMBOLS "compile with debug symbols" OFF)
+option(DEBUG_NO_OPTIMIZE "do not optimize the code" OFF)
+
+if (DEBUG_OUTPUT)
+  add_definitions(-DDEBUG)
+endif()
+
 add_definitions(-DVERSION="${gitver}")
 
+function(set_compiler_flags TARGETS)
+  foreach(TARGET IN LISTS TARGETS)
+    if ( CMAKE_C_COMPILER_ID MATCHES "Clang|AppleClang|GNU")
+      target_compile_options(${TARGET} PRIVATE -Wall -Wextra -Wunreachable-code)
+    endif()
+    if ( CMAKE_C_COMPILER_ID MATCHES "Clang")
+      target_compile_options(${TARGET} PRIVATE -Wweak-vtables -Wexit-time-destructors -Wglobal-constructors -Wmissing-noreturn )
```

Comment:
Quite a useless set of warnings for a pure C code, I'd say. Why do you want them added?